feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/following/FollowingFragment.java

```diff
@@ -1,33 +1,176 @@
+import com.example.feelslikemonday.ui.login.SignupActivity;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import com.baoyz.swipemenulistview.SwipeMenu;
+import com.baoyz.swipemenulistview.SwipeMenuCreator;
+import com.baoyz.swipemenulistview.SwipeMenuItem;
+
+/**
+ * Responsible for listing all followees' moods and related information
+ */
 
-/*Responsible for listing all followers' moods and related information */
 public class FollowingFragment extends Fragment {
 
     private FollowingViewModel followingViewModel;
+    private SwipeMenuListView userList;
+    private FolloweeBookAdapter adapter;
+    private List<String> followeeUsernames;
+    private static FollowPermissionDAO DAO;
+    private SharedPreferences pref;
+    private String myUserID;
+    private List<MoodEvent> followeeUserMoodList;
+    private ArrayList<followeeMoodEvent> myfolloweeList = new ArrayList<>();
+    private int xIteration;
+    private int userVisited = 0;
+    private String  userCurrent;
 
-    public View onCreateView(@NonNull LayoutInflater inflater,
-                             ViewGroup container, Bundle savedInstanceState) {
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         followingViewModel = ViewModelProviders.of(this).get(FollowingViewModel.class);
         View root = inflater.inflate(R.layout.fragment_following, container, false);
         followingViewModel.getText().observe(getViewLifecycleOwner(), new Observer<String>() {
+
             @Override
             public void onChanged(@Nullable String s) {
-                //do stuff here
+
+                pref = getActivity().getApplicationContext().getSharedPreferences(SignupActivity.PREFS_NAME, 0);
+                myUserID = pref.getString(SignupActivity.USERNAME_KEY,null);
+                userList = getView().findViewById(R.id.followee_list);
+
+                DAO = FollowPermissionDAO.getInstance();
+                DAO.get(myUserID, new FollowPermissionCallback(){
+                    @Override
+                    public void onCallback(FollowPermission followPermission) {
+                        if(followPermission.getFollowerUsername().equals(myUserID)){
+
+                            followeeUsernames = followPermission.getFolloweeUsernames();
+
+                            UserDAO userDAO = new UserDAO();
+
+                            for (xIteration=0; xIteration<followeeUsernames.size(); xIteration++){
+                                 userCurrent = followeeUsernames.get(xIteration);
+                                 userDAO.get(userCurrent, new UserCallback() {
+                                    @Override
+                                    public void onCallback(User user) {
+
+                                        checkFollowee(user);
+
+                                        userVisited = userVisited + 1;

```

Comment:
Why is this an attribute and not a variable? userVisited will not reset to zero at any point. Is this intentional?

Reply:
youre right, fixed